### PR TITLE
Removing dependency of syncprompt #66

### PR DESCRIPTION
**Pull request by jah2488**
A relatively simple PR that removes `syncprompt` in favor of using node's built in `readline` library.

- There were a handful of other places I can see where improvements could be made, but decided to attempt and keep this PR more focused.
- The majority of the diff is actually caused by PrettierJS, happy to clean that out of the PR and make be _only_ my changes if needed.

I had issues running the tests with `yarn test`, but if I ran the code found in `yarn test` manually, everything passed fine.

### CODE REVIEW

File: upgrade.js [hunk 1]
```diff
@@ -250,46 +252,55 @@ function main(knownPackages) {
...
         );
+      }
 
+      rl.question("[Y/n]: ", (answer) => {
+        logMessage("\n");
+        var proceed = yn(answer);
+        if (proceed) {
+          updatePackages();
+        }
+        rl.close();
```
Comment (jah2488) [inline upgrade.js]: The majority of the changes are here.

**Comment by avh4** [comment]
Cool, yeah looks like the CI failure is because of prettier.. I usually just fix it with `npx prettier --write upgrade.js`

**Comment by avh4** [comment]
Looks good!  Forgot to remove syncprompt from package.json, though

**Comment by jah2488** [comment]
@avh4 Its been a long day 😅

**Comment by avh4** [comment]
Awesome, thanks!!

**Pull request merged**